Optional List in SectionConfiguration throws NullPointerException

```java
//SkriptEvent
private final SectionConfiguration configuration = new SectionConfiguration.Builder().addOptionalList("aliases").build();
```
```java
@Override
public List<Statement> loadSection(FileSection section, ParserState parserState, SkriptLogger logger) {
	configuration.loadConfiguration(null, section, parserState, logger);
	Optional.ofNullable(configuration.getStringList("aliases")).ifPresent(list -> Arrays.stream(list).forEach(aliases -> this.aliases.add(aliases)));
	return configuration.getSection("trigger").getItems();
}
```
Then when calling:
```
@Override
public List<Statement> loadSection(FileSection section, ParserState parserState, SkriptLogger logger) {
	Optional.ofNullable(configuration.getStringList("aliases"))
       		.ifPresent(list -> Arrays.stream(list).forEach(aliases -> this.aliases.add(aliases)));
}
```
```
command /test:
	#aliases: no aliases node
	trigger:
		broadcast "testing"
```
Throws this:
```
[Server thread/ERROR] #!#! Caused by: java.lang.NullPointerException: Cannot invoke "Object.getClass()" because "result" is null
[Server thread/ERROR] #!#!     at io.github.syst3ms.skriptparser.lang.entries.SectionConfiguration.getValue(SectionConfiguration.java:135)
[Server thread/ERROR] #!#!     at io.github.syst3ms.skriptparser.lang.entries.SectionConfiguration.getStringList(SectionConfiguration.java:145)
```

--- a/src/main/java/io/github/syst3ms/skriptparser/lang/entries/LiteralLoader.java
+++ b/src/main/java/io/github/syst3ms/skriptparser/lang/entries/LiteralLoader.java
@@ -38,7 +38,7 @@ public class LiteralLoader<T> extends OptionLoader {
 		if (isMultiple()) {
 			List<T> data = new ArrayList<>();
 			boolean successful = true;
-			for (var value : config.getStringList(key)) {
+			for (var value : config.getStringList(key).orElse(new String[0])) {
 				var result = parser.apply(value);
 				if (result == null) {
 					// With the logic that errors get skipped, we will allow the other values to be parsed.
@@ -52,7 +52,7 @@ public class LiteralLoader<T> extends OptionLoader {
 			config.getData().put(key, data.toArray());
 			return successful;
 		} else {
-			var result = parser.apply(config.getString(key));
+			var result = parser.apply(config.getString(key).orElseThrow());
 			// We don't want this data to linger if an error occurs
 			config.getData().remove(key);
 			if (result == null) {

--- a/src/main/java/io/github/syst3ms/skriptparser/lang/entries/SectionConfiguration.java
+++ b/src/main/java/io/github/syst3ms/skriptparser/lang/entries/SectionConfiguration.java
@@ -12,6 +12,7 @@ import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 
 /**
  * The SectionConfiguration class can be used to create simple and efficient data sections
@@ -124,28 +125,44 @@ public class SectionConfiguration {
 	 * This can only be used when you register your option as a {@link Builder#addLiteral(String, Class) literal},
 	 * otherwise, the option will likely be parsed as a String and throw an exception.
 	 * Options that allow literal lists are saved as an array.
+	 * Returns an empty Optional if the key is not present. This is only possible for optional keys.
+	 *
 	 * @param key the key
 	 * @param cls the class to cast to
-	 * @return the value cast to the given class
+	 * @return the value cast to the given class, or an empty Optional if the key was not specified
 	 * @param <T> the type of the value
 	 */
 	@SuppressWarnings("unchecked")
-	public <T> T getValue(String key, Class<T> cls) {
+	public <T> Optional<T> getValue(String key, Class<T> cls) {
 		var result = data.get(key);
+		if (result == null)
+			return Optional.empty();
 		if (result.getClass() == String.class && result.getClass() != cls)
 			throw new UnsupportedOperationException("The key '" + key + "' was not registered as a literal, was parsed as a String and can, therefore, not be cast to " + cls.getName());
-		return (T) result;
+		return Optional.of((T) result);
 	}
 
-	public String getString(String key) {
+	/**
+	 * @param key the key
+	 * @return the option value, or an empty Optional if the key was not specified
+	 */
+	public Optional<String> getString(String key) {
 		return getValue(key, String.class);
 	}
 
-	public String[] getStringList(String key) {
+	/**
+	 * @param key the key
+	 * @return the list values, or an empty Optional if the key was not specified
+	 */
+	public Optional<String[]> getStringList(String key) {
 		return getValue(key, String[].class);
 	}
 
-	public CodeSection getSection(String key) {
+	/**
+	 * @param key the key
+	 * @return the enclosed code section, or an empty Optional if the key was not specified
+	 */
+	public Optional<CodeSection> getSection(String key) {
 		return getValue(key, CodeSection.class);
 	}
 	

--- a/src/test/java/io/github/syst3ms/skriptparser/syntax/test/SecCategory.java
+++ b/src/test/java/io/github/syst3ms/skriptparser/syntax/test/SecCategory.java
@@ -45,8 +45,9 @@ public class SecCategory extends CodeSection {
     @Override
     public Optional<? extends Statement> walk(TriggerContext ctx) {
         Variables.setVariable("the_number", config.getValue("number"), null, false);
-        Variables.setVariable("multiple", String.join(";", config.getStringList("multiple")), null, false);
-        return Optional.of(config.getSection("die"));
+        Variables.setVariable("multiple", String.join(";", config.getStringList("multiple").orElseThrow()), null, false);
+        config.getString("optional").ifPresent(s -> Variables.setVariable("optional", s, null, false));
+        return Optional.of(config.getSection("die").orElseThrow());
     }
 
     @Override

--- a/src/test/resources/general/categories.txt
+++ b/src/test/resources/general/categories.txt
@@ -1,6 +1,8 @@
 # Author(s):
 # 	- Mwexim
-# Date: 2021/11/23
+# Date:
+#	- 2021/11/23
+#   - 2024/01/01 (NullPointerException fix)
 
 test:
 	birth:
@@ -28,3 +30,6 @@ test:
 
 	# In the code, it sets this variable to the joined values
 	assert {multiple} is equal to "This is the first value;This is the second value" with "{multiple} has not been set correctly: %{multiple}%"
+
+	# The 'optional' option was not specified, so in the code, this variable should not be set
+	assert {optional} is not set with "{optional} has been set when it shouldn't have been: %{optional}%"

